### ISSUE TITLE
Fingerprint elements are generated randomly

### ISSUE THREAD

**Issue by simon-nowak** [GridexX/sarif-codeclimate#1]
Hello here,

I started using the sarif-codeclimate project for our own CI. I'm really happy with it. Thank you for this work.

I think I found a bug. The `fingerprint` field is filled with a random hash. 

As a result when a MR is trying to compare 2 branchs and do a diff report then the report show an already existing error as new. And the old one as "fixed"

![image](https://user-images.githubusercontent.com/35262768/200373041-dbf00032-b142-4f1b-819d-21270e5e46e3.png)

**Current behaviour** 
Considere this current result in a SARIF report 

```json
"results": [
    {
    "level": "error",
    "ruleId": "ClassNamingConventions",
    "ruleIndex": 0,
    "message": {
        "text": "The test class name 'cit_apexTest' doesn't match '[A-Z][a-zA-Z0-9_]*'"
    },
    "locations": [
        {
        "physicalLocation": {
            "artifactLocation": {
            "uri": "file:///Users/xxx/workspace/salesforce-ci/citest/force-app/main/default/classes/cit_apexTest.cls"
            },
            "region": {
            "startLine": 2,
            "endLine": 6,
            "startColumn": 28,
            "endColumn": 2
            }
        }
        }
    ]
    }
]
```

After conversion 
```json
    {
        "description": "The test class name cit_apexTest' doesn't match '[A-Z][a-zA-Z0-9_]*'",
        "fingerprint": "6d6a37738b2ea8605fe70a3a977ba21b",
        "severity": "major",
        "location": {
            "path": "force-app/main/default/classes/cit_apexTest.cls",
            "lines": {
                "begin": 2
            }
        }
    },
```

The "fingerpring" tag will be different on each run.

**Expected behaviour** 
When an error is found with the same message at the same line, it should generate the same hash.

If we agree about the issue I might be able to propose an MR to fix it.

**Comment by GridexX** [comment]
Hello @simon-nowak !

Thank you for using my project :smile: 


Of course, contributions are welcome !
Indeed, we agree about the issue, it shoudl be changed

:point_right: The code should be updated in `lib/parser.js` to use a md5 generated from the description instead of a random one :

```javascript
//Generate a random md5 hash for the fingerprint
const randomMD5 = () => crypto.createHash('md5').update(Math.random().toString()).digest('hex');
```

Good luck for your mission !

Don't hesitate to star the project it if you like it, it would help me a lot :+1:

**Comment by GridexX** [comment]
@simon-nowak Hello, what's the state of your work ? Did you start to work on an MR ?

**Comment by simon-nowak** [comment]
Hello, sorry for the late answer. No I had to put that on a side. 

edit: I'm considering closing this because of the new version. Should I?

**Comment by GridexX** [comment]
No problem, in fact it had not been fixed in this version.  You will find the function in the `lib/parser.ts` file

**Comment by GridexX** [comment]
@simon-nowak I have fixed this issue in the `2.1.1` version of the package.
Could you give it a try and tell me if the problem was solved for your project

**Comment by simon-nowak** [comment]
It seems ok to me.

Thank you for your amazing work !
